test/libsolidity/syntaxTests/scoping/name_shadowing_function_parameter_name_vs_contract.sol

```diff
@@ -0,0 +1,20 @@
+interface I {
+    function f(uint I) external;         // OK
+}
+
+library L {
+    function f(uint L) public pure {}    // warning
+}
+
+abstract contract A {
+    function f(uint A) public pure {}    // warning
+    function g(uint A) public virtual;   // OK
```

Comment:
Why is this "OK"? Shadowing still exists right?

Reply:
See https://github.com/ethereum/solidity/pull/10908#issuecomment-781355831.

Having a parameter named like a type inside a function declaration that has no body cannot break anything so we do not consider that shadowing. It's clear what it's referring to. You don't want to warn about it and annoy the user. When there is a body, on the other hand, warning makes sense.